### ISSUE TITLE
Budget alerts are not being set for the projects created in 0-bootstrap step

### ISSUE THREAD

**Issue by romanini-ciandt** [terraform-google-modules/terraform-example-foundation#1042]
### TL;DR

I'm opening this issue TBD because I'm not sure if the behavior I'll describe is intentional or not.

In 0-bootstrap step, we have the creation of 2 projects (`prj-b-seed` and `prj-b-cicd-random`). These projects aren't being created by a direct call to [Google's project-factory module](https://registry.terraform.io/modules/terraform-google-modules/project-factory/google/latest) like most of the project creation we have at foundation. I mean, under the hoods it is created by Google's project-factory module, but it's **an indirect call** made by other modules.

The creation of `prj-b-seed` project is through [Google's bootstrap module](https://registry.terraform.io/modules/terraform-google-modules/bootstrap/google/latest) in [0-bootstrap main.tf file](https://github.com/terraform-google-modules/terraform-example-foundation/blob/56450bdd16ca178b4f1191aadc3690b45cbf8f3d/0-bootstrap/main.tf#L45) and the creation of `prj-b-cicd-random` project is through a submodule of this same [Google's bootstrap module](https://registry.terraform.io/modules/terraform-google-modules/bootstrap/google/latest) at [0-bootstrap cb.tf file](https://github.com/terraform-google-modules/terraform-example-foundation/blob/56450bdd16ca178b4f1191aadc3690b45cbf8f3d/0-bootstrap/cb.tf#L79).

Given that context, let me introduce the possible issue: When we have direct calls to [Google's project-factory module](https://registry.terraform.io/modules/terraform-google-modules/project-factory/google/latest) we provide some `budget_alert` inputs (`budget_amount`, `budget_alert_pubsub_topic`, etc) for the project-factory module that configures budget alarms, which nowadays is not happening for the projects created in 0-bootstrap step. **So because of that, we won't have any budget alarm for the projects created in 0-bootstrap.**

### Expected behavior

The following code is a [snippet from 1-org](https://github.com/terraform-google-modules/terraform-example-foundation/blob/56450bdd16ca178b4f1191aadc3690b45cbf8f3d/1-org/envs/shared/projects.tf#L57) (inside foundation) where we have a project being created.
Note that we have inputs for budget_alert.

``` terraform
module "org_billing_logs" {
  source  = "terraform-google-modules/project-factory/google"
  version = "~> 14.0"

  random_project_id        = true
  random_project_id_length = 4
  default_service_account  = "deprivilege"
  name                     = "${local.project_prefix}-c-billing-logs"
  org_id                   = local.org_id
  billing_account          = local.billing_account
  folder_id                = google_folder.common.id
  activate_apis            = ["logging.googleapis.com", "bigquery.googleapis.com", "billingbudgets.googleapis.com"]

  labels = {
    environment       = "production"
    application_name  = "org-billing-logs"
    billing_code      = "1234"
    primary_contact   = "example1"
    secondary_contact = "example2"
    business_code     = "abcd"
    env_code          = "p"
  }
  budget_alert_pubsub_topic   = var.project_budget.org_billing_logs_alert_pubsub_topic
  budget_alert_spent_percents = var.project_budget.org_billing_logs_alert_spent_percents
  budget_amount               = var.project_budget.org_billing_logs_budget_amount
}
```

### Observed behavior

The following code is a [snippet from Google's bootstrap module](https://github.com/terraform-google-modules/terraform-google-bootstrap/blob/8dcf43b6561a85cd3e867003ef167ea222d87403/modules/tf_cloudbuild_source/main.tf#L33) (outside foundation).
Note that budget alerts are not being defined. This is also true for cicd project.

``` terraform
module "seed_project" {
  source                      = "terraform-google-modules/project-factory/google"
  version                     = "~> 14.0"
  name                        = local.seed_project_id
  random_project_id           = var.random_suffix
  disable_services_on_destroy = false
  folder_id                   = var.folder_id
  org_id                      = local.seed_org_depends_on
  billing_account             = var.billing_account
  activate_apis               = local.activate_apis
  create_project_sa           = false
  labels                      = var.project_labels
  lien                        = true
}
```

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```
